config/confighttp/confighttp.go

```diff
@@ -147,6 +158,17 @@ func (hcs *HTTPClientSettings) ToClient(host component.Host, settings component.
 
 	transport.DisableKeepAlives = hcs.DisableKeepAlives
 
+	if hcs.HTTP2ReadIdleTimeout != nil && hcs.HTTP2ReadIdleTimeout.Seconds() > 0 {
+		transport2, err := http2.ConfigureTransports(transport)
+		if err != nil {
+			return nil, fmt.Errorf("failed to configure http2 transport: %w", err)
+		}
+		transport2.ReadIdleTimeout = *hcs.HTTP2ReadIdleTimeout
+		if hcs.HTTP2PingTimeout != nil {
```

Comment:
these are ignored if the idle timeout isn't also set, which functionally is the same tmk but may be more reflective to configure an http2 transport if either value isn't nil.

Reply:
@rmfitzpatrick 
> these are ignored if the idle timeout isn't also set

are you referring to the "read idle timeout" ? if yes, these comes together, in other words, if `ReadIdleTimeout` is set to 0 or not set, we should not set the ping timeout as the health check will not be performed

Reply:
>we should not set the ping timeout 

This is the flow of the config application I'm referring to, but it may be a matter of preference. If the aim is to "respect user input" then it'd make sense to set the values specified and leave the semantics to the underlying transport.

Reply:
same as previous comment, the idea why adding the extra call, ie `ConfigureTransports` , if knowing that it will be ignored

Reply:
likely covered by https://github.com/open-telemetry/opentelemetry-collector/pull/9022#discussion_r1412504732, but because the user specified it